### ISSUE TITLE
Y24-099 Tube Rack Banking Sequencescape API

### ISSUE THREAD

**Issue by yoldas** [sanger/limber#1696]
**User story**
As PSD, we would like to update Sequencescape v2 API to provide Limber support for TubeRack model similar to Plates using the versioned tube rack model proposed in Reuse of TubeRacks document.

**Who are the primary contacts for this story**
Abdullah, Katy, Andrew

**Who is the nominated tester for UAT**
TBD

**Acceptance criteria**
- [x] Check existing labware creator (in  DPL-811) for Sequencescape API calls and replace them for tube racks.
- [x] Check TubeRacks and associated objects are retrieved by using JSONAPI includes, i.e. Tubes and RackedTube.
- [x] Add resource methods to TubeRacks and RackedTubes so that they are similar to Plate and Wells.
- [x] Add TubeRack barcode into new TubeRack poly_metadata in order to enable history tracking of physical tube rack

**Dependencies**
This story is blocked by the following dependencies:
- #1482

References
This story has a non-blocking relationship with:
- #1314
- #1713

**Additional context**
The versioned tube rack model is researched in DPL-1011 Re-use of tube racks and the model is summarised in the Google Document[ DPL-1011 Re-use of tube racks](https://docs.google.com/document/d/1rvPXeCk5rBEf72VkchHCD2FRmWANboke2pln_Lt_j5U/edit?usp=sharing)

This story is part of re-implementation of [DPL-811 Transfer PBMC isolations into FluidX tube racks](https://github.com/sanger/limber/issues/1314) (Banking) using the versioned tube rack model.

Story relationships are shown in the Lucidchart diagram [TubeRack Stories](https://lucid.app/lucidchart/d88bdcfd-9628-4fec-8b5a-453f680ee06b/edit?viewport_loc=-1051%2C-316%2C2770%2C1486%2C0_0&invitationId=inv_13d74286-1cc2-4c2f-aaf0-30887764601c) .

It is not possible to release the implementation of this story in isolation.

**Comment by TWJW-SANGER** [comment]
A couple of queries:

- Do the features that require these APIs already exist or need to created? I suspect "API Support to allow History tracking of physical tube rack.." is a feature that does not yet exist?  Would it make sense to split that feature out into its own _user_ story so that we understand what the business need before designing this API call?
- I can see "API Support to find a tube on a tube rack" would be useful / needed. Curious as to how it works at the moment and/or where this is being called from. In the sense of - could this be split out into another smaller story?

**Comment by KatyTaylor** [comment]
"API Support to allow History tracking of physical tube rack using poly_metadata" - a bit more detail would be good. I think what we mean here is that the tube rack barcode will be saved into poly_metadata when the tube rack is created - is that correct?

**Comment by yoldas** [comment]
TODO: Add more detail to how the criteria can be done if possible.

**Comment by KatyTaylor** [comment]
1. spec/models/labware_creators/plate_split_to_tube_racks_spec.rb fix breaking tests **- part of this story**
2. understand / fix lots of other broken tests on PR into epic branch **- part of this story**
3. see if we can refactor the labware creator code into common tube rack includes modules **- leave until Cryostor work, when new labware creator will be needed**
4. tube rack state - currently have a method on TubeRackPresenter for state, move to model? causes bug in relatives tab code if not present **- put this into the presenter story.** 
5. submission of tube racks - is this needed? currently using tube submission 'Limber-Htp - scRNA Core cDNA Prep GEM-X 5p' on the tubes, and then scan them in for cryostor creation **- part of the cryostor block of stories - write it down somewhere on those. If submission is on tube rack, need a way to create a tube rack before making the submission.**
6. driver file for Bank plate to tubes is not working 'Hamilton LRC PBMC Bank plate to LRC Bank Seq and Spare tubes CSV' **- story to fix this** 
7. bed verification is not working 'Hamilton LRC PBMC Bank => LRC Bank Seq and LRC Bank Spare' **- have existing story**
8. manual transfer not working - integration suite cannot progress without bed verification OR 'Manual Transfer' **- make story to fix this**
9. tube custom metadata for rack and coordinate - needed? **- this is OK, we want this to be removed - racked_tubes should replace this**
10. Think about the 'redoing the step' scenario & re-jig the existing backlog stories for the new data model.
11. Check on work completion code - is it triggered, test it after manual transfer is done. Relates to tube rack state (#4) - new story
12. Fix integration suite **- make sure written down in acceptance criteria. Create new epic branch and merge Andrew's branch in.**
